Update Installation Docs

BowTiedDevOps opened the pull request:
Updated docs and scripts to install bitcoin v25.0 and stacks-blockchain v2.4.0.1.0.

BowTiedDevOps committed:
feat: update docs to match bitcoin 25.0 installation

Also match stacks-blockchain v2.4.0.1.0

bitnet-io commented:
for bitcoin v26 bitcoin.conf

deprecatedrpc=create_bdb

bitnet-io commented:
https://github.com/bitnet-io/stacks-core-bitnet/releases/tag/Stacks-2.0-mining-for-Bitcoin-people

bitnet-io commented:
should update the website as soon as possible so people arent using instructions that are impossible

https://docs.stacks.co/stacks-in-depth/nodes-and-miners/mine-mainnet-stacks-tokens

BowTiedDevOps commented:
This change is PRd to testnet branch, and it applies to that network. It needs further testing and possibly improvements, but I will update the mainnet and the documentation as soon as I can. Thank you for the v26.0 tip!

BowTiedDevOps committed:
fix: add missing `\` in bitcoin wallet creation

wileyj commented on bitcoin.md:
```diff
@@ -18,27 +18,37 @@ curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/stacksfou
+$ export CXXFLAGS="-O2"
+$ ./configure \
+  CXX=clang++ \
+  CC=clang \
```
is clang required now? i want to try this for myself....

wileyj replied:
i do have a local script to build bitcoin that uses clang....but i'm curious if it's *required* now. have to test myself - will resolve if that's true. 

if it is, we may need to adjust the doc a little with some recommended dependency install

wileyj commented on bitcoin.md:
```diff
@@ -18,27 +18,37 @@ curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/stacksfou
- --without-miniupnpc \
- --with-pic \
- --enable-cxx \
- --with-boost-libdir=/usr/lib/x86_64-linux-gnu
```
i would be surprised if you don't need this. will test.

BowTiedDevOps replied:
Worked without

wileyj replied:
confirmed, this seems fine and `libminiupnpc` is a default package now

wileyj commented on bitcoin.md:
````diff
@@ -72,14 +82,14 @@ EOF'
-```
+```bash
 $ sudo bash -c 'cat <<EOF> /etc/systemd/system/bitcoin.service
 [Unit]
````
not able to properly suggest a change here, but i'd like to add in some conditions to the unit:
```
$ sudo bash -c 'cat <<EOF> /etc/systemd/system/bitcoin.service
[Unit]
Description=Bitcoin daemon
After=network.target
ConditionFileIsExecutable=/usr/local/bin/bitcoind
ConditionPathExists=/bitcoin
ConditionPathExists=/etc/bitcoin/bitcoin.conf
ConditionFileNotEmpty=/etc/bitcoin/bitcoin.conf
```

double check my suggestion, but i think that would work

BowTiedDevOps replied:
`ConditionPathExists` and `ConditionFileNotEmpty` point to the same file. Is the prior required, or can we only keep the latter instead of checking the same file twice? Other than that looks like a good addition.

wileyj replied:
I think using `ConditionFileNotEmpty` by itself should be good enough, but check the systemd docs to verify

BowTiedDevOps replied:
Just tested and committed, it works only with that

BowTiedDevOps replied:
P.S. also added it for stacks (the config part, the `FileIsExecutable` and datadir `PathExists` were already there)

BowTiedDevOps replied:
Citing systemd [docs](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#:~:text=ConditionFileNotEmpty%3D,non%2Dzero%20size.):
```
ConditionFileNotEmpty=
ConditionFileNotEmpty= is similar to ConditionPathExists= but verifies that a certain path exists and refers to a regular file with a non-zero size.
```

wileyj commented on prerequisites.md:
```diff
@@ -60,30 +60,37 @@ The following packages are required, and used by the rest of these docs
-    jq \
-    netcat \
-    nodejs \
+    bsdmainutils \
```
this is also interesting - is this required for the changes in how BDB is built?

BowTiedDevOps replied:
Not required, removed it

wileyj commented on scripts/install_stacks.sh:
```diff
@@ -19,7 +19,7 @@ git clone --depth 1 --branch master https://github.com/stacks-network/stacks-blo
 cargo build --features monitoring_prom,slog_json --release --bin stacks-node
 sudo cp -a "${HOME}/stacks-blockchain/target/release/stacks-node" "/usr/local/bin/stacks-node"
-
+sudo rm -rf "${HOME}/stacks-blockchain"
```
```suggestion
```

i would keep the source code on the system for rebuilds

BowTiedDevOps replied:
Understood, will remove the remove

wileyj commented on scripts/install_stacks.sh:
```diff
@@ -131,8 +145,8 @@ ConditionFileIsExecutable=/usr/local/bin/stacks-node
-ExecStart=/bin/sh -c "/usr/local/bin/stacks-node start --config=/etc/stacks-blockchain/Config.toml >> /stacks-blockchain/miner.log 2>&1"
-ExecStartPost=/bin/sh -c "umask 022; sleep 2 && pgrep -f \"/usr/local/bin/stacks-node start --config=/etc/stacks-blockchain/Config.toml\" > /run/stacks-blockchain/stacks.pid"
+ExecStart=/bin/sh -c "/usr/local/bin/stacks-node start --config /etc/stacks-blockchain/Config.toml >> /stacks-blockchain/miner.log 2>&1"
+ExecStartPost=/bin/sh -c "umask 022; sleep 2 && pgrep -f \"/usr/local/bin/stacks-node start --config /etc/stacks-blockchain/Config.toml\" > /run/stacks-blockchain/stacks.pid"
```
this is ok for now - but i think we can reduce the complexity of this unit with changes to the shutdown process of the stacks-node binary:
https://github.com/stacks-network/stacks-core/blob/master/contrib/init/stacks.service

wileyj replied:
resolving. this is good enough for now, i'll create an issue to update so it's closer to https://github.com/stacks-network/stacks-core/blob/master/contrib/init/stacks.service

wileyj commented on stacks-blockchain.md:
```diff
@@ -15,6 +15,7 @@ $ git clone --depth 1 --branch master https://github.com/stacks-network/stacks-b
 $ cd $HOME/stacks-blockchain/testnet/stacks-node
 $ cargo build --features monitoring_prom,slog_json --release --bin stacks-node
 $ sudo cp -a $HOME/stacks-blockchain/target/release/stacks-node /usr/local/bin/stacks-node
+$ sudo rm -rf $HOME/stacks-blockchain
```
see above comment

wileyj commented on scripts/prerequisites.sh:
```diff
@@ -20,30 +20,37 @@ curl -sL https://deb.nodesource.com/setup_16.x | sudo -E bash -
-    jq \
-    netcat \
-    nodejs \
+    bsdmainutils \
```
see above comment

BowTiedDevOps replied:
Not required, removed it

wileyj commented:
> create_bdb

will have to test this on v26, but for v25 it doesn't seem to be required. can you create an issue for this so we can track it when we get to testing v26?

wileyj commented:
overall looks good - few changes requested that should be simple to resolve.

BowTiedDevOps committed:
feat: do not remove stacks-blockchain dir after compiling

BowTiedDevOps committed:
feat: systemd conditions for processes, remove `bsdmainutils`

wileyj commented:
@BowTiedDevOps one more change i came across here: can you update the package iinstall for `netcat`, renaming to `ncat`?

most recent version of debian uses the name `ncat` whereas previous versions allowed `netcat`

BowTiedDevOps committed:
feat: update `netcat` to `ncat`

BowTiedDevOps committed:
feat: mine_microblocks false, update bitcoin build prefix

wileyj commented on scripts/install_stacks.sh:
```diff
@@ -52,28 +51,42 @@ STX_ADDRESS=$(sudo cat  /root/keychain.json | jq .keyInfo.address | tr -d '"')
     echo "[ install_stacks.sh ] - Sleeping for 120 seconds to allow bitcoin time to restart"
     sleep 120
 
+    echo "[ install_stacks.sh ] - Loading bitcoin wallet"
+    bitcoin-cli \
+        -rpcconnect=127.0.0.1 \
+        -rpcport=18332 \
+        -rpcuser=btcuser \
+        -rpcpassword=btcpass \
+        loadwallet miner
```
no longer required
```suggestion
```

wileyj commented on wallet.md:
```diff
@@ -37,24 +37,36 @@ _Import will only be successful after bitcoin has fully synced_
   -rpcuser=btcuser \
   -rpcpassword=btcpass \
-importmulti '[{ "scriptPubKey": { "address": "<npx btcAddress>" }, "timestamp":"now", "keys": [ "<npx wif>" ]}]' '{"rescan": true}'
+  loadwallet miner
```
this command starting on line 52 is no longer required

wileyj commented:
sharing here from our discord chat:
`importmulti` will work to import an address, however if you're importing an older bitcoin address the timestamp will need to be adjusted or you won't load the address utxos. 

since the docs show how to create a new address, i think a note is a good compromise, i.e.
```
if you're importing an old address, consider changing the timestamp in the importmulti command, or use importprivkey 
<links to cli docs>
```

BowTiedDevOps committed:
feat: remove bitcoin restart and loadwallet calls

- add bitcoin wallet import on startup option in config
- option to import and use existing btc address in bitcoin wallet

BowTiedDevOps committed:
feat: remove wallet_dir from config

wileyj reviewed (approved):
👍
